client/lib/jetpack/paths.ts

```diff
@@ -36,6 +36,6 @@ export const socialBasePath = () => '/jetpack-social';
 export const socialPath = ( siteSlug?: string ): string =>
 	siteSlug ? `${ socialBasePath() }/${ siteSlug }` : socialBasePath();
 
-export const partnerPortalBasePath = () => '/partner-portal';
+export const partnerPortalBasePath = ( path = '' ) => `/partner-portal${ path }`;
```

Comment:
Awesome addition! 👏 

*It might be outside the scope of this PR and is purely a question instead of an actual request:*
Since we register all paths in `client/jetpack-cloud/sections/partner-portal/index.ts`, could there be a win in creating a function that would function something like the following:

```js
export const getPartnerPortalPath = ( page = '',  args = '' ) => {
  const basePath = partnerPortalBasePath();

  // The switch would allow us to modify any special rules where the "page name" doesn't match the path.
  switch( page ) {
    case 'licenses':
      return `${ basePath }/get-licenses${ args }`;

    default:
      return `${ basePath }/${ path }${ args }`;
  }
}
```

I've written the function here in GitHub and it doesn't include any "registered pages" logic, but the point being that we have a single place where we request a URL instead of hardcoding e.g. `partnerPortalBasePath( '/licenses' )` then we would be able to call `getPartnerPortalPath( 'licenses' )` and if we ever change  the `licenses` to be located at `get-licenses` (very theoretical example) then we would only have to update the path in `getPartnerPortalPath` and keeps the code DRY

Again: this is a question(!) and could easily just be an over-engineered idea for a fictional problem that we will never reach.